File: mdoc/src/main/scala/mdoc/internal/worksheets/EvaluatedWorksheet.scala
package mdoc.internal.worksheets

import java.{util => ju}
import mdoc.{interfaces => i}
import java.nio.file.Path

case class EvaluatedWorksheet(
    val diagnostics: ju.List[i.Diagnostic],
    val statements: ju.List[i.EvaluatedWorksheetStatement],
    val files: ju.List[i.ImportedScriptFile],
    val scalac: ju.List[String],
    val classpath: ju.List[Path],
    val dependencies: ju.List[coursierapi.Dependency],
    val repositories: ju.List[coursierapi.Repository],
    val exportableEvaluation: ju.Optional[String]
) extends mdoc.interfaces.EvaluatedWorksheet

File: mdoc-interfaces/src/main/scala/mdoc/interfaces/EvaluatedWorksheet.java
package mdoc.interfaces;

import java.util.List;
import java.util.Optional;
import java.nio.file.Path;
import coursierapi.Dependency;
import coursierapi.Repository;

public abstract class EvaluatedWorksheet {

  public abstract List<Diagnostic> diagnostics();
  public abstract List<EvaluatedWorksheetStatement> statements();

  public abstract List<ImportedScriptFile> files();
  public abstract List<String> scalac();
  public abstract List<Path> classpath();
  public abstract List<Dependency> dependencies();
  public abstract List<Repository> repositories();
  public abstract Optional<String> exportableEvaluation();
}
